### ISSUE TITLE
New Job contruct is not dealing correctly with the environment array

### ISSUE THREAD

**Issue by julbrs** [serverless-stack/sst#2089]
# How to reproduce ?

Add a `Job` construct like that:

```
  const job = new Job(stack, "Job", {
    handler: "src/job.main",
    environment: {
      FOO: "value",
    },
  });
```

Then use `process.env.FOO` in your hander function.

# Actual behavior

The env. var is not defined.

# Expected behavior

We want to have the `FOO` env. var. defined.

# More information

* I am on SST `1.15.1`
* The function `job.addEnvironment("FOO", "hello");` is working! 

Thanks !

**Comment by fwang** [comment]
Fixing it

**Comment by julbrs** [comment]
OMG 
I opened this case at 2pm and a PR is already merged at 5pm. That's terrific 🤩

BTW The `Job` construct is working well for my today use case (sync data between 2 systems, take a few hours and SQS not very easy to set up because I don't want to deal with API rate limit issues)

Thanks !

**Comment by fwang** [comment]
Haha, thanks @julbrs. It's released in v1.15.2
